src/utils/unpack.cairo

```diff
@@ -0,0 +1,50 @@
+use array::ArrayTrait;
+use traits::Into;
+use traits::TryInto;
+use option::OptionTrait;
+use eternum::constants::RESOURCE_IDS_PACKED_SIZE;
+use eternum::constants::PRIME;
+use traits::BitAnd;
+use eternum::utils::math::pow;
+use debug::PrintTrait;
+
+fn unpack_resource_ids(resource_ids_packed: u128, resource_ids_count: u8) -> Array<u8> {
```

Comment:
that was my original plan but i ran into some withdraw_gas errors when running tests and couldn't get rid of it. since withdraw_gas shouldn't be added anymore to the code and adding it does not remove the error, i'm thinking these are bugs that might get resolved in future cairo versions. will can try again in future milestones if theres more unpacking to do

Reply:
Not sure if related, but I ran into a similar issue today on a different project. The "fix" was to split the test function into a couple of smaller ones, then the error went away 🤷 

FWIW, the error was:

```sh
Error: Failed setting up runner.

Caused by:
    Failed calculating gas usage, it is likely a call for `gas::withdraw_gas` is missing.
```

Reply:
yeah we can tweak in next milstone, lets not block here on some upstream thing